modules/core/arrow-core/src/main/kotlin/arrow/core/Function1.kt

```diff
@@ -32,5 +34,7 @@ class Function1<I, out O>(val f: (I) -> O) : Function1Of<I, O> {
     }
 
     fun <I, A, B> tailRecM(a: A, f: (A) -> Function1Of<I, Either<A, B>>): Function1<I, B> = { t: I -> step(a, t, f) }.k()
+
+    fun <I> id(): Function1<I, I> = Function1(::identity)
```

Comment:
I used the `Function1` constructor explicitly because the compiler
could not infer the type of `::identity` when I used `::identity.k()`